src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3266,14 +3266,14 @@ private static bool TryGetNumberValue(string strNum, bool hex, bool real, char s
                             {
                                 d = -0.0;
                             }
-                            if (suffix == 'l' || suffix == 'L')
+                            if (suffix == "l" || suffix == "L")
                             {
                                 result = ((long)Convert.ChangeType(d, typeof(long), CultureInfo.InvariantCulture)) * multiplier;
                             }
-                            if (suffix == 'u' || suffix == 'U') {
+                            if (suffix[0] == 'u' || suffix[0] == 'U') {
                                 ulong testresult = ((ulong)Convert.ChangeType(d, typeof(ulong), CultureInfo.InvariantCulture)) * (ulong)multiplier;
 
-                                if (testresult > UInt32.MaxValue) {
+                                if (testresult > UInt32.MaxValue || String.Compare(suffix, "ul", true) == 0) {
```

Comment:
We avoid this overload in favor of the one taking `StringComparison.OrdinalIgnoreCase`.